### PR TITLE
Fixing#565

### LINKED ISSUE #565: [BUG] Missing declaration for getPointOnCurve/getDirectionOfCurve
**Describe the bug**
The part of the function declaration is missing. This bug corresponds to functions getPointOnCurve and getDirectionOfCurve for the entities:

- [x]  IfcThirdOrderPolynomialSpiral #566 
- [x] IfcSecondOrderPolynomialSpiral #551
- [x] IfcSeventhOrderPolynomialSpiral #552 
- [x] IfcSine (resolved in commit [0461e5e](https://github.com/tumcms/Open-Infra-Platform/pull/561/commits/0461e5e0e74f78dc572057b4253960ed3e5c4e12))
- [x] IfcCosine #561

```
template <>
carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcThirdOrderPolynomialSpiral>& thirdOrderPolynomial,
const typename IfcEntityTypesT::IfcParameterValue& parameter) const noexcept(false)
{
 return getPointOnCurve(thirdOrderPolynomial, parameter * this- >UnitConvert()->getLengthInMeterFactor());
}
```
Without this part the code never reaches functions `getPointOnCurve/getDirectionOfCurve` and calculates points with catresianPointList.

**To Reproduce**
1. Go to 'CurveConverter'
2. Scroll down to the line 2755
3. See declarations

**Expected behavior**
When rendering each type of geometry, we should not see the message on the command windows that corresponding geometry is not supported yet.

**CMake configuration**
If applicable, list any special configurations you have selected in CMake.
- EARLYBINDING_WITH_IFC4X3_RC4: selected

### PR DESCRIPTION
**Pull request by Elvira2227**
Mistake in function declaration fixed #565 
Corresponding information in documentation added in `IfcThirdOrderPolynomialSpiral`

### CODE REVIEW

File: Core/src/IfcGeometryConverter/CurveConverter.h [hunk 1]
```diff
@@ -2678,6 +2678,12 @@ namespace OpenInfraPlatform
 				* \note
 				*/
 				template <>
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcThirdOrderPolynomialSpiral>& thirdOrderPolynomial,
+					const typename IfcEntityTypesT::IfcParameterValue& parameter) const noexcept(false)
+				{
+					return getPointOnCurve(thirdOrderPolynomial, parameter * this->UnitConvert()->getLengthInMeterFactor());
+				}
+				template <>
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/CurveConverter.h:2686]: I can't select the lines below in the review, so:

Line 2692/2693 - the second function parameter should be `double` and not `IfcParameterValue`.

**Review by pjanck — CHANGES_REQUESTED** [review]

**Review by pjanck — APPROVED** [review]
lftm

Thank you for finding another case.

File: Core/src/IfcGeometryConverter/CurveConverter.h (hunk 1 above)
Comment (Elvira2227) [reply Core/src/IfcGeometryConverter/CurveConverter.h:2686]: Resolved in [bf6802c](https://github.com/tumcms/Open-Infra-Platform/pull/566/commits/bf6802c2e85df55ec447ff0f2c7f663b4dd121ba)

**Review by jschlenger — APPROVED** [review]
Looks fine

**Pull request merged**